### DISCUSSION TITLE
[BP#55] Privilégier les changements visuels instantanés

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#211] [Bonnes pratiques] [answered]
N'est-ce pas un cas particulier de la [BP #43](https://github.com/cnumr/best-practices/blob/main/chapters/BP_043_fr.md) ?

Peut-être à mettre en exemple de la 43 qui est très générique.

**Comment by ACTLEM** [comment]
Cela peut être vu comme un cas particulier, après ici cela insiste plus pour éviter les animations. J'avoue que les effets "animés" m'énervent, mais surtout ralentisse l'utilisation du service car attirent l'attention. Je serais plus pour la garder mais curieux d'autres avis.

**Comment by florinesueur** [comment]
Oui en effet, elle est liée à la [BP#43](https://github.com/cnumr/best-practices/blob/main/chapters/BP_043_fr.md) : Réduire au maximum le repaint (appearence) et le reflow (layout) mais aussi à la [BP#39](https://github.com/cnumr/best-practices/blob/main/chapters/BP_039_fr.md) : Éviter les animations JavaScript / CSS coûteuses, la [BP#42](https://github.com/cnumr/best-practices/blob/main/chapters/BP_042_fr.md) : Rendre les éléments du DOM invisibles lors de leur modification, la [BP#45](https://github.com/cnumr/best-practices/blob/main/chapters/BP_045_fr.md) : Modifier plusieurs propriétés CSS en 1 seule fois et la nouvelle BP en discussion #30 : Compartiments CSS qui peut-être comme une alternative (ou plutôt un moins pire avec une action limitée).

Je suis pour l'animation des snackbars, snipers, ... tous systèmes d'alerte, de changement d'état ou d'aide qui doivent attirer l'attention de l'utilisateur. Bien sûr avec utilité et modération.

J'ai l'impression que réduire au maximum le repaint (appearence) et le reflow (layout) est l'objectif et nous donnons quelques BP pour y arriver comme limiter les animations et transformations, faire les changements en 1 seule fois ...

**Reply by devcrafting** [reply]
Tu laisserais l'objectif sous forme de BP quand même ? Ou uniquement les BP plus concrètes ? Ou intégrer les exemples concrets dans la BP globale

**Comment by Fred-artwai** [comment]
Au delà de çà, la partie animation comme l'exemple de cette BP ne devrait pas être en JS mais en CSS. Les animations doivent être réservés aux CSS. Je pense (mais je peux me tromper) que le JS est beaucoup plus gourmand que le CSS. Du coup, un changement visuel, doit se traduire par un simple changement de class ou d'attribut en JS pour correspondre à un selecteur CSS qui va déclencher une animation ou une transition CSS. 

L'objet de cette BP est-elle de proscrire les animations (qui sont parfois nécessaires pour des raisons de lisibilité ou d'ergonomie) ?
ou de proscrire les animations JS ? Et là je dis oui. 

Je me demande même si doit pas faire l'objet d'une nouvelle BP.

**Comment by ACTLEM** [comment]
On garde cette BP en supprimant la référence au CSS ou on se dit finalement que c'est une redite des BP 39, 42 et 43 (donc on la supprime)?

**Reply by florinesueur** [reply]
Je me demande si nous sommes exhaustif sur les solutions pour éviter la BP43 globale. A défaut d'être exhaustif, il faut alors garder la BP globale mais possiblement faire référence aux différentes BP plus concrètes qui peuvent être mises en place en tant qu'exemples et solutions alternatives.

Du coup, si je suis cette logique je garderai cette BP indépendante au même titre que les autres.

**Comment by ACTLEM** [comment, marked as answer]
J'ai créé l'issue pour amélioration éventuelle (suppression CSS?): https://github.com/cnumr/best-practices/issues/314
